parse-server-ascension/cloud/student.js

```diff
@@ -187,20 +187,41 @@ Parse.Cloud.define("AssignHouse", async(request) => {
     const query = new Parse.Query(House); 
     const argument = request.params;
     const res = await query.find();
-    
-    //Get lowest!
-    var randNum = Math.floor(Math.random() * res.length);
-    var pickedHouse = res[randNum];
 
+    //Note: All houses should begin with the same population count; base case (e.g [0,0,0,0])
+    const res2 = JSON.parse(JSON.stringify(res))
+    let max = 0
+    let counter = 0
+    for(let i = 0; i < res2.length; i++){
+        if(max < res2[i].HousePopulation){
+            max = res2[i].HousePopulation
+            counter++
+        }
+    }
+    //counter == 0 => houses have same population [1,1,1,1]
+    //counter == 1 => houses have 2 different populations [1,0,0,1]
+    //cuonter > 1 => houses have 3 or more different populations [1,3,0,1]
+    if(counter == 1){
+        for(let i = 0; i < res2.length; i++){
+            if(res2[i].HousePopulation == max){
+                res2.splice(i, 1)
```

Comment:
Not sure pag splice digdi, baka may ma miss out na index, safe na mag decrement ning i paka splice, syempre check mo rin baka mag negative si i.
Ignore nalang kung working naman.

**Safer/Efficient solution?** 
Instead na max, min nalang hanapon then insert the similar values in a new array, O(n) kasi si splice, and O(1) lang mag push

Reply:
Suggestion ko lang yan last comment haha